pkg/controllers/nodetemplate/infrastructure.go

```diff
@@ -67,6 +68,10 @@ func (i *InfrastructureReconciler) Reconcile(ctx context.Context, nodeTemplate *
 	} else if len(list.Items) >= 1 {
 		if i.lastInfrastructureReconcile.Add(time.Minute * 5).Before(time.Now()) {
 			if err := i.CreateInfrastructure(ctx); err != nil {
+				if errors.IsRecentlyDeleted(err) {
```

Comment:
can we mitigate this by using some randomness in the queue name?

Reply:
We rely on the queue name. It's not tag-based, it's static so we can't add randomness to the name.